### PR TITLE
Added support to collection for named urls

### PR DESCRIPTION
**Pull request by john-westcott-iv**
##### SUMMARY
With this change awx.awx colleciton modules can used named URLs. For example:
```
---
- name: Test a named URL
  hosts: localhost
  connection: local
  gather_facts: False
  tasks:
    - name: Create a job template
      awx.awx.job_template:
        name: named_url_test
        job_type: "run"
        inventory: "Demo Inventory++Default"
        project: "Demo Project++Default"
        playbook: "hello_world.yml"
        credentials:
          - "Demo Credential++Machine+ssh++"
        state: "present"
```

##### ISSUE TYPE

 - New or Enhanced Feature

##### COMPONENT NAME

 - Collection

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by AlanCoding** [comment]
Organizations are vanilla name-unique. So you would have to do `organization: Default++`? I don't know if this is useful or intended...

**Comment by sean-m-sullivan** [comment]
Vanilla names should still work, What I've been trying to do is make it so it takes the output of export out of the box, and using that as the metric, if we could an output for export to put out the named urls... well that would be /chefskiss

**Comment by ericzolf** [comment]
It might make sense to make it clear that `get_one` remains the entry point by prefixing the two new functions with underscore? i.e. `_get_by_name` and `_get_by_named_url`

**Comment by john-westcott-iv** [comment]
I found a bug playing around with this and will be providing an update to the code.

**Comment by john-westcott-iv** [comment]
@AlanCoding items with just a single named reference (i.e. Orgs) would skip the named reference task in this current implementation. Since a named reference is a pretty simple git we could remove the check on `++` and always try the GET for a named reference first and then fall back to the lookup method. This should be much easier in the new implementation of the code. Let me know if you think its worth it, I'm currently debating if it would be "faster" since we already say to use "Name" but that would only apply to named lookups where they only have one part for the name so maybe not?

**Comment by AlanCoding** [comment]
How should we document features like this? We have plugin-level docs with examples, but because this is general that's not effective. For stuff like authentication, there are parameters that correspond to features so those options can be documented on their own.

To me that leaves the README. Maybe we need a general feature overview that covers stuff like this. Otherwise, how do users find it?

**Comment by john-westcott-iv** [comment]
@AlanCoding I did open an issue for updating all of the docs to say: "name, ID or named URL" #14208. Which would help but yea, we could also do something in the root readme.

**Comment by john-westcott-iv** [comment]
I put a place holder in the readme which will need a last minute replacement once we know what version this will be released with.

File: awx_collection/README.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ Notable releases of the `awx.awx` collection:
  - 7.0.0 is intended to be identical to the content prior to the migration, aside from changes necessary to function as a collection.
  - 11.0.0 has no non-deprecated modules that depend on the deprecated `tower-cli` [PyPI](https://pypi.org/project/ansible-tower-cli/).
  - 19.2.1 large renaming purged "tower" names (like options and module names), adding redirects for old names
+ - X.X.X added support of named URLs to all modules. Anywhere that previously accepted name or id can also support named URLs
```
Comment (AlanCoding) [inline awx_collection/README.md:71]: so it's either going to be 22.5.1 or 22.6.0 or 23.0.0?

File: awx_collection/plugins/module_utils/controller_api.py [hunk 2]
```diff
@@ -421,7 +443,6 @@ def get_one(self, endpoint, name_or_id=None, allow_none=True, check_exists=False
             self.fail_wanted_one(response, endpoint, new_kwargs.get('data'))
 
         if check_exists:
-            name_field = self.get_name_field_from_endpoint(endpoint)
```
Comment (AlanCoding) [inline awx_collection/plugins/module_utils/controller_api.py:424]: so surprised no linter could pick this up

**Review by AlanCoding — APPROVED** [review]
This seems complete, I just want to look for some feedback that people see this and use it and it doesn't mess them up.

**Review by jay-steurer — APPROVED** [review]

**Pull request merged**